server/src/main/java/org/opensearch/identity/PermissionResult.java

```diff
@@ -0,0 +1,19 @@
+package org.opensearch.identity;
+
+/**
+ * Results of permissions checks
+ * */
+public interface PermissionResult {
```

Comment:
I'm missing the purpose of this interface.  A boolean + error message (presumably if false) seems to probably exist somewhere more generically, e.g., `BooleanResultWithErrorMessage`, or at least as an object rather than an interface if associated only with `Permission`. Is there a potential for more to be added to this?  I'm not seeing much diversity in implementing classes here.

It seems that it should be context dependent, e.g., the result for a specific `Permission` (or list of Permissions).

Reply:
It is pretty limited and so I see you point on its limited utility, but I wanted to force a focal point for the evolution in the design to hold more interesting information.

For example, results boil down to allowed/denied, what about a request that has 3 permissions associated with it, overall we might want it to be seen as denied, but maybe a non-binary outcome needs to be handled.  Batching this together as a single permissions result seems worthwhile when the performance of roundtrips is a factor.